### DISCUSSION TITLE
[React] Popover arrow aligned center for position

### DISCUSSION THREAD

**Discussion by DmitryShcheklein** [floating-ui/floating-ui#2358] [Q&A] [answered]
Hello! I need help.

I'm trying to create Popover with arrow on react.
I use [Reusable popover component]
https://floating-ui.com/docs/popover
Than i move to:
https://floating-ui.com/docs/floatingarrow#usage
But doc example FloatingArrow isn't full for prev example, it is too simple..

Now i have code:
https://codesandbox.io/s/compassionate-tereshkova-2kuisy?file=/src/Popover.tsx:6902-6919
`const staticSide` not work. How i can aligned an arrow?

![Снимок экрана 2023-06-01 в 20 06 41](https://github.com/floating-ui/floating-ui/assets/16781081/61caa101-e628-4bfc-b025-eec1c8021fe4)
![Снимок экрана 2023-06-01 в 20 07 33](https://github.com/floating-ui/floating-ui/assets/16781081/d4e26f04-cade-4540-8c68-b97410c85713)

**Comment by atomiks** [comment, marked as answer]
You're calling `usePopover()` again inside `PopoverContent`, but you're meant to `usePopoverContext()` instead. And it gets automatically positioned by the component, so you don't need to use the positioning styles.

This works:

```js
export const PopoverContent = React.forwardRef<
  HTMLDivElement,
  React.HTMLProps<HTMLDivElement>
>(function PopoverContent(props, propRef) {
  const { context: floatingContext, arrowRef, ...context } = usePopoverContext();
  const ref = useMergeRefs([context.refs.setFloating, propRef]);

  if (!floatingContext.open) return null;

  return (
    <FloatingPortal>
      <FloatingFocusManager context={floatingContext} modal={context.modal}>
        <div
          ref={ref}
          // className={styles.content}
          style={{
            ...context.floatingStyles,
            ...props.style
          }}
          aria-labelledby={context.labelId}
          aria-describedby={context.descriptionId}
          {...context.getFloatingProps(props)}
        >
          <FloatingArrow
            ref={arrowRef}
            context={floatingContext}
            fill="red"
          />
          {props.children}
        </div>
      </FloatingFocusManager>
    </FloatingPortal>
  );
});
```

**Reply by DmitryShcheklein** [reply]
@atomiks, thanks!
